cms/browse/views.py

```diff
@@ -8,9 +9,25 @@
 logger = logging.getLogger("general")
 
 
+def fetch_page_by_slug(slug):
+    ###########
+    # https://trello.com/c/47ZFPaJb/255-fix-non-unique-slugs-in-browse-pages
+    # This should be a unique page slug
+    # programme_page = Page.objects.get(slug=programme)
+    # Also see browse_tags.py:~58
+    #######
+    # But we have multiples, so we need to guard against that.
+    pages = Page.objects.filter(slug=slug)
+    if len(pages) == 0:
+        # Should not be reachable
+        raise Http404
+    return list(pages)[0]
```

Comment:
In an ideal world we would make sure every page has a unique slug, but I'm worried that given how deep the site tree goes this could inadvertently create a lot of very strange URLs (eg `/department-1/what-we-do` and `/department-2/what-we-do` would collide, so we might end up with `department-1/what-we-do` and `department-2/what-department-2-does`). I'm not inherently against this forcing of uniqueness, but the behaviour wants checking.

Reply:
I'll add a warn, and we have a ticket to look at slugs in the back log. The import process is littered with duplicates.

Reply:
Oh, so each individual fragment at any level has to be unique? That feels horrid to me. Definately worth a closer look.